rust/src/node_manager/cached_node_manager.rs

```diff
@@ -304,16 +347,110 @@ mod tests {
         types::{NodeId, NodeType},
     };
 
+    type TestCachedNodeManager = CachedNodeManager<NodeId, Node, MockCachedNodeManagerStorage>;
+    const ONE_MB: usize = 1024 * 1024;
+
     #[test]
     fn cached_node_manager_new_creates_node_manager() {
+        let capacity = ONE_MB;
+        let storage = MockCachedNodeManagerStorage::new();
+        let manager = TestCachedNodeManager::new(capacity, storage);
+        let total_node_size: usize = (Node::min_non_empty_node_size()
+            + mem::size_of::<RwLock<NodeWithMetadata<Node>>>())
+            * manager.nodes.len();
+        assert!(
+            total_node_size
+                + ((size_of::<NodeId>() + 21 + size_of::<usize>())
+                    * ((manager.nodes.len() - 1) as f64 * 1.5) as usize)
+                    .next_power_of_two()
+                <= capacity
+        );
+    }
+
+    #[test]
+    #[should_panic]
+    fn cached_node_manager_new_fails_if_capacity_too_small() {
```

Comment:
```suggestion
    fn cached_node_manager_new_panics_if_capacity_too_small() {
```
We use `fails` often times when we mean that it returns an error.